deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -219,6 +219,11 @@ public Response create(
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("This stage does not allow deploying a private build. Please Contact #teletraan and #security-related to whitelist your stage for deploying private build"));
         }
         
+        // check if the build is from trusted artifact url
+        if(!buildBean.getArtifact_url().contains("deployrepo.pinadmin.com") && ! buildBean.getEnsure_trusted_build()) {
```

Comment:
the trusted_url should be configured in the config file. 
the enforcement should be checking the environBean.ensure_trusted_build, if ensure_trusted, the artifact_url need to be from trusted_url.  probably should use prefix instead of contains.

Reply:
Can I ask where is this "trusted_url" set?

Reply:
could be similar to https://github.com/pinterest/teletraan/pull/692 and the config is in server.yaml. if the config is not set, could ignore the check. for non-opensource config, it is in server-prod.yaml or server-integ.yaml

Reply:
Thanks for explaining, currently I added 2 checks 1) invalid url check 2) untrusted url check. Please review.